tests/unit/test_validation.py

```diff
@@ -66,7 +66,34 @@ def test_validate_payload_with_single_object_schema_and_multiple_valid_args(
         )
 
 
-def test_validate_payload_with_array_schema_and_multiple_valid_args(
+def test_validate_payload_with_array_schema_and_single_valid_arg(
+    faker: Faker,
+):
+    validate_payload(
+        args=[faker.pylist(value_types=[str])],
+        schema={
+            "type": "array",
+            "items": {"type": "string"},
+        },
+    )
+    assert True
+
+
+def test_validate_payload_with_array_schema_and_single_invalid_arg(
+    faker: Faker,
+):
+    with pytest.raises(PayloadValidationException):
+        validate_payload(
+            args=[faker.pylist(value_types=[int])],
```

Comment:
I think for this to pass the `args` needs to be a `Sequence` with the 0th element being the actual array.
`([1,2,3,4], )`

Reply:
This is because of how `validate_payload` is called, it is getting passed the `*args` from the event handler. Which in the tuple case is `(1, 2, 3)` but in the array case is `([1,2,3], )`

Reply:
This test may benefit from a larger test using the `publish_message_validator_factory` and other validator factories. to better simulate the condition or even an integration style test that passes an actual message to cover how socketIO calls the handler.

Reply:
@alex-zywicki Have changed all the args to be tuples rather than lists. Good shout.

With regards to testing this on the integration level, I think it is an overkill. The `validate_payload` method is the single point in the code base the implements/affects this behaviour. Covering this function as a unit suffices IMO.

Reply:
Actually, thinking about it again, I think an integration test asserting both cases would be of benefit. Adding it.